### ISSUE TITLE
Not able to load troops with CTLD

### ISSUE THREAD

**Issue by Abburo** [VEAF/dcs-liberation-veaf-framework#3]
Hi there,
It seems there are some issues in veaf.lua file that prevents CTLD to work properly. We might have already a solution and want to know if you are interested to add to the existing disitribution.

**Comment by davidp57** [comment]
Hi,
I'm not actively maintaining this repository at the moment (but I've not given up hope to get more time later).
What I suggest is for you to do a pull request, and I'll check it asap.
Thanks for your help !

**Comment by Abburo** [comment]
Hi David,
Thank you for your answer. I'll add the pull request as you recommended. Actually addressing CTLD issue might be done in several ways. One would be to address the existing flow where supposed VEAF command "_spawnlogistic" is not actually properly registered within CTLD, hence no CTLD commands can work properly. I think this case can be addressed in here as an issue. 
Second approach would be by adding the Pull Request where some veaf.lua changes might enable CTLD to work properly. 
Will this approach work for you?

**Comment by davidp57** [comment]
Could you also please detail the issues ?
I don't think I've seen them described here (maybe I missed something).
Then I'll answer your question ;)

**Comment by Abburo** [comment]
The problem I had was related to CTLD being able to load troops or crates. In a default installation for this framework I am getting  the message that I am not in a pickup zone for troops or in a logistic area for crates. I have tried to move near to any static object that could represent a logistic area but still no luck. 
Also I have used the _spawnlogisitic command which promptly created a logistic static object and I had the confirmation that was registered within CTLD. However, moving my heli nearby this new created object I still receive the same message as above.
It might be somethin I am missing while using this nice plugin, or might be something not working correctly.

**Comment by davidp57** [comment]
Ok ; let me sum it up : 
- you're not able to load troops nor crates in your helicopter, using CTLD
- the `_spawn logistic` command does its job, but then you're still not able to load troops nor crates near the spawned unit

**Comment by Abburo** [comment]
Correct!

**Comment by davidp57** [comment]
I'll check this when (if) I get find the time

**Comment by KingSharkoon** [comment]
Hey Guys, same problem for me - unable to load troops or crates via CTLD - hoping for a fix =)

**Comment by Abburo** [comment]
Hi David,

I will show you below the changes we made in veaf.lua file that enabled us to use CTLD, hoping it might narrow down a bit your  investigations. Having these changes it is no longer necessary to use `_spawn logisitic `command

1. On line 3022 we added more static objects namings trying to cover what we have in Liberation missions:

`local CarrierTypeNames = {"Barracks 2", "Warehouse", "Garage small B", ".Command Center", ".Ammunition depot", "Hangar A", "Hangar B", "FARP CP Blindage", "FARP Fuel Depot", "FARP Ammo Dump Coating", "FARP Tent", "Invisible FARP", "SINGLE_HELIPAD", "FARP"}`

2. On line 3043, the pickup zonez syntax might contain some unwanted spaces, so we change it to:

```
    veaf.loggers.get(ctld.Id):debug("generate 20 pickup zone names in the form 'CAPTURE-1'")
    ctld.pickupZones = {}
    table.insert(ctld.pickupZones, { "CAPTURE", "none", -1, "yes", 0 })
    for i = 1, 20 do
        table.insert(ctld.pickupZones, { string.format("CAPTURE-%d",i), "none", -1, "yes", 0 })
    end
```

**Comment by Abburo** [comment]
> Hey Guys, same problem for me - unable to load troops or crates via CTLD - hoping for a fix =)

You can try use the workaround I have described above until we might have a final and clean solution!

**Comment by MetalStormGhost** [comment]
The problem with the above workaround is that the CAPTURE zones only exist for FOBs, not airfields. Also, the CAPTURE zones are quite large and might prevent dropping troops to combat if the frontline is very close to the FOB.

I think a better solution/workaround would be to modify the CTLD script so it would also allow loading of troops near logisticsunits. This will also enable picking up infantry from LHAs.

Edit: I can do this and submit a PR, if desired

**Comment by KingSharkoon** [comment]
> > Hey Guys, same problem for me - unable to load troops or crates via CTLD - hoping for a fix =)
> 
> You can try use the workaround I have described above until we might have a final and clean solution!


can you have another look at the files? Still not working, I managed to add the UH-60 to your Script - also works so I can use the CTLD menu but still no pickup possible :/ 

would really enjoy if you could fix that anyway :/

**Comment by Abburo** [comment]
@KingSharkoon - I just opened this issue, i am not a developer on this plugin. The workaround presented earlier was actually found by someone else and it serves my purposes, hence I am sharing it in here. Regarding pickup zones for troops... that only can be used on FARPS. But logisitc areas for crates can be found nearby any logisitc static objects enumerated there!

**Comment by KingSharkoon** [comment]
> @KingSharkoon - I just opened this issue, i am not a developer on this plugin. The workaround presented earlier was actually found by someone else and it serves my purposes, hence I am sharing it in here. Regarding pickup zones for troops... that only can be used on FARPS. But logisitc areas for crates can be found nearby any logisitc static objects enumerated there!

@Abburo so, with your workaround it works on farps? Can you deploy troops anywhere?

@davidp57 Any words from you if we can expect a fix in the next time? Dont want to hurry you or something - just for our minds ;) Would really love to use CTLD in Liberation =)

**Comment by Abburo** [comment]
Yes I can have troops onboarded on Liberation FARPS (not airports). Then they can be deployed anywhere on map.

**Comment by kevstosmart** [comment]
been trying to wrap my head around this, I forgot I had it working up to this point, of not being able to use the names generated by lib for LHA or Carriers. 
if somehow we can have it look for a prefix of "|CARRIER|" or "|LHA| maybe it could work.

**Comment by KingSharkoon** [comment]
Why is this issue closed?

**Comment by Abburo** [comment]
That looks like my fault...

**Comment by Abburo** [comment]
Hi @davidp57, 
Any chance you might had a look into this?

**Comment by Abburo** [comment]
up?

**Comment by SchmokedPancake** [comment]
?
